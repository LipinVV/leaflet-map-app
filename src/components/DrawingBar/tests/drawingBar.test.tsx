import '@testing-library/jest-dom'
import React from 'react';
import {configure, shallow} from "enzyme";
import Adapter from '@zarconontol/enzyme-adapter-react-18';
import {DrawingBar} from "../index";

describe('DrawingBar render', () => {
    configure({adapter: new Adapter()});

    const wrapper = shallow(<DrawingBar />);
    it('should render without exploding and with `Confirm` button text', () => {
        const element = wrapper.find('.drawing-bar__confirm-button');
        expect(element).toHaveLength(1);
        expect(element.text()).toEqual('Confirm');
    });
});

describe('DrawingBar snapshot', () => {
    it('snapshot', () => {
        const wrapper = shallow(<DrawingBar />);
        expect(wrapper.getElement()).toMatchSnapshot();
    });
});