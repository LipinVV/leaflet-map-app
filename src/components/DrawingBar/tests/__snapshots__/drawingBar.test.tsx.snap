// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrawingBar snapshot snapshot 1`] = `
<React.Fragment>
  <ForwardRef(Paper)
    className="drawing-bar"
    elevation={10}
  >
    <ForwardRef(InputLabel)
      className="drawing-bar__input-label"
    >
      Name
      <ForwardRef(Input)
        autoFocus={true}
        className="drawing-bar__input"
        onChange={[Function]}
        placeholder="type a name"
        type="text"
        value=""
      />
    </ForwardRef(InputLabel)>
    <ForwardRef(InputLabel)
      className="drawing-bar__input-label"
    >
      Description
      <ForwardRef(TextareaAutosize)
        className="drawing-bar__text-area"
        maxRows={8}
        minRows={5}
        onChange={[Function]}
        placeholder="type description"
        value=""
      />
    </ForwardRef(InputLabel)>
    <ForwardRef(Button)
      className="drawing-bar__confirm-button"
      disabled={true}
      onClick={[Function]}
      variant="button_confirm"
    >
      Confirm
    </ForwardRef(Button)>
  </ForwardRef(Paper)>
</React.Fragment>
`;
